google/cloud/bigquery/job.py

```diff
@@ -2067,14 +2067,23 @@ def destination_uri_file_counts(self):
     def to_api_repr(self):
         """Generate a resource for :meth:`_begin`."""
 
+        configuration = self._configuration.to_api_repr()
         source_ref = {
             "projectId": self.source.project,
             "datasetId": self.source.dataset_id,
-            "tableId": self.source.table_id,
         }
 
-        configuration = self._configuration.to_api_repr()
-        _helpers._set_sub_prop(configuration, ["extract", "sourceTable"], source_ref)
+        if isinstance(self.source, TableReference):
+            source_ref["tableId"] = self.source.table_id
+            _helpers._set_sub_prop(
+                configuration, ["extract", "sourceTable"], source_ref
+            )
```

Comment:
These lines are very similar in both variants. I'd be setting a variable with `"sourceTable"/"sourceModel"` value, and then one big call:
```
_helpers._set_sub_prop(
    configuration, ["extract", source], source_ref
)
```